python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py

```diff
@@ -69,30 +69,43 @@ class AutoMaterializePolicy(
 ):
     """An AutoMaterializePolicy specifies how Dagster should attempt to keep an asset up-to-date.
 
-    There are two main kinds of auto-materialize policies: eager and lazy. In essence, an asset with
-    an eager policy will try to immediately materialize after upstream changes, while an asset with
-    a lazy policy will only materialize when necessary in order to satisfy the relevant
-    FreshnessPolicies.
+    Each policy consists of a set of `AutoMaterializeRule`s, which are used to determine if there is
+    some reason that an asset / partition of an asset should be auto-materialized, and if so, if
+    there is some reason that an asset / partition of an asset should not be auto-materialized.
 
-    For an asset / partition of an asset with an _eager_ policy to be auto-materialized, at least
-    one of the following must be true:
+    The most common policy is `AutoMaterializePolicy.eager()`, which consists of the following rules:
```

Comment:
I see you replaced the rule descriptions with the actual in-code rules, but I see value in keeping both. Thoughts on adding the original descriptions to their new in-code counterparts here? Ex:

> - `AutoMaterializeRule.materialize_on_missing()` - An asset or a partition of an asset is missing

Thoughts?